src/CMakeLists.txt

```diff
@@ -24,3 +24,27 @@ set_target_properties(read_parquet
         LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
         RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin
 )
+
+add_executable(simple_thread_pool)
+target_sources(simple_thread_pool
+    PRIVATE
+        enum.hpp
```

Comment:
We should move common files to a library to avoid re-including them in every project.  If you submit this for now, then I'll refactor and move executables to an app folder